tests/test_api.rs

```diff
@@ -8,27 +7,6 @@ use jsonschema::{Draft, JSONSchema, SchemaResolverError};
 use serde_json::Value;
 use url::Url;
 
-// rust doesn't have post-test hooks so this is to allow creating temporary
-// files that get removed when they go out of scope
-struct LocalTempFile {
-    file_name: String
-}
-
-impl LocalTempFile {
-    fn new(file_name: &str, body: &[u8]) -> LocalTempFile {
-        let mut file = fs::File::create(file_name).expect("File created");
-        file.write_all(body).expect("File written");
-        LocalTempFile { file_name: String::from(file_name) }
-    }
-}
-
-impl Drop for LocalTempFile {
```

Comment:
Didn't realise tests run in parallell so modifying the example directory by adding/removing files in one test was causing failures of other tests. Have just backed this out and comitted invalid files to the test directory to test error paths.